File: backend/ongil/src/main/java/kr/co/ongil/domain/patient/health/repository/HealthDataRepository.java
package kr.co.ongil.domain.patient.health.repository;

import kr.co.ongil.domain.patient.health.entity.HealthData;
import kr.co.ongil.domain.patient.health.entity.HealthDataType;
import org.springframework.data.domain.Sort;
import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDateTime;
import java.util.List;

/**
 * 건강 데이터 레포지토리
 */
public interface HealthDataRepository extends JpaRepository<HealthData, Integer> {

    /**
     * 환자 ID와 측정 시각 범위로 조회 (모든 타입)
     */
    List<HealthData> findByPatientIdAndMeasuredAtBetween(
        Integer patientId,
        LocalDateTime from,
        LocalDateTime to,
        Sort sort
    );

    /**
     * 환자 ID, 데이터 타입, 측정 시각 범위로 조회
     */
    List<HealthData> findByPatientIdAndTypeAndMeasuredAtBetween(
        Integer patientId,
        HealthDataType type,
        LocalDateTime from,
        LocalDateTime to,
        Sort sort
    );

    /**
     * 환자 ID와 데이터 타입으로 최신 데이터 조회
     */
    HealthData findFirstByPatientIdAndTypeOrderByMeasuredAtDesc(
        Integer patientId,
        HealthDataType type
    );

    /**
     * 환자 ID로 모든 건강 데이터 삭제 (환자 삭제 시)
     */
    void deleteAllByPatientId(Integer patientId);
}

File: android/ongil/app/src/main/java/kr/co/ongil/domain/repository/HealthDataRepository.kt
package kr.co.ongil.domain.repository

import kr.co.ongil.data.model.health.HealthDataResponse
import kr.co.ongil.data.model.health.HealthSummaryResponse
import kr.co.ongil.domain.model.HealthData
import kr.co.ongil.domain.model.HealthDataType

/**
 * 헬스 데이터 원격 Repository 인터페이스
 */
interface HealthDataRepository {
    /**
     * 생체 데이터 업로드
     * @param patientId 환자 ID
     * @param healthData 업로드할 건강 데이터
     * @return Result<String> 성공 메시지 또는 에러
     */
    suspend fun uploadHealthData(patientId: Long, healthData: HealthData): Result<String>

    /**
     * 생체 데이터 조회
     * @param patientId 환자 ID
     * @param type 조회할 데이터 종류 (null이면 전체)
     * @param from 조회 시작 날짜 (yyyyMMdd)
     * @param to 조회 종료 날짜 (yyyyMMdd)
     * @param sort 정렬 기준 (기본: measuredAt,desc)
     * @return Result<HealthDataResponse> 건강 데이터 조회 결과
     */
    suspend fun getHealthData(
        patientId: Long,
        type: HealthDataType? = null,
        from: String? = null,
        to: String? = null,
        sort: String = "measuredAt,desc"
    ): Result<HealthDataResponse>

    /**
     * 생체 데이터 요약 통계 조회 (일별 단위)
     * @param patientId 환자 ID
     * @param type 통계할 데이터 종류 (null이면 전체)
     * @param from 조회 시작 날짜 (yyyyMMdd)
     * @param to 조회 종료 날짜 (yyyyMMdd)
     * @return Result<HealthSummaryResponse> 건강 데이터 요약 통계
     */
    suspend fun getHealthDataSummary(
        patientId: Long,
        type: HealthDataType? = null,
        from: String? = null,
        to: String? = null
    ): Result<HealthSummaryResponse>

    /**
     * 생체 데이터 삭제
     * @param patientId 환자 ID
     * @param healthDataId 삭제할 건강 데이터 ID
     * @return Result<String> 성공 메시지 또는 에러
     */
    suspend fun deleteHealthData(
        patientId: Long,
        healthDataId: Long
    ): Result<String>
}